scripts/generate-helm-config-for-hcs.sh

```diff
@@ -0,0 +1,110 @@
+#!/usr/bin/env bash
+
+############################################################################
+# WARNING: This script is experimental and is not meant for production use #
+############################################################################
+#
+# Script to retrieve configuration from Hashicorp Consul Service on Azure,
+# bootstrap ACLs, and create Kubernetes secrets and Helm config file
+# to install this Helm chart.
+
+set -euo pipefail
+
+: "${subscription_id?subscription_id environment variable required}"
+: "${resource_group?resource_group environment variable required}"
+: "${managed_app_name?managed_app_name environment variable required}"
+: "${cluster_name?cluster_name environment variable required}"
+
+echo "-> Fetching cluster configuration from Azure"
+cluster_resource=$(az resource show --ids "/subscriptions/${subscription_id}/resourceGroups/${resource_group}/providers/Microsoft.Solutions/applications/${managed_app_name}/customconsulClusters/${cluster_name}" --api-version 2018-09-01-preview)
+cluster_config_file_base64=$(echo "${cluster_resource}" | jq -r .properties.consulConfigFile)
+ca_file_base64=$(echo "${cluster_resource}" | jq -r .properties.consulCaFile)
+
+echo "Writing cluster configuration to consul.json"
+echo "${cluster_config_file_base64}" | base64 --decode | jq . > consul.json
+
+echo "Writing CA certificate chain to ca.pem"
+echo "${ca_file_base64}" | base64 --decode > ca.pem
+echo
+
+echo "-> Bootstrapping ACLs"
+
+# Extract the URL for the servers.
+# First, check if the external endpoint is enabled and if yes, use the external endpoint URL.
+# Otherwise, use the private endpoint URL.
+external_endpoint_enabled=$(echo "${cluster_resource}" | jq -r .properties.consulExternalEndpoint)
+if [ "$external_endpoint_enabled" == "enabled" ]; then
+  server_url=$(echo "${cluster_resource}" | jq -r .properties.consulExternalEndpointUrl)
+else
+  server_url=$(echo "${cluster_resource}" | jq -r .properties.consulPrivateEndpointUrl)
+fi
+
+# Call Consul bootstrap API and save the bootstrap secret
+# to a Kubernetes secret if successful.
+output=$(curl -sX PUT "${server_url}"/v1/acl/bootstrap)
+if grep -i "permission denied" <<< "$output"; then
+  echo "ACL system already bootstrapped."
```

Comment:
I totally missed this in my output. Can you use terminal colour chars to make his red.

Reply:
Even better would be if this prints at the end of the script.

Reply:
Added colors.